controllers/progressiverollout_controller_test.go

```diff
@@ -22,6 +25,26 @@ const (
 	interval = time.Millisecond * 10
 )
 
+type mockArgoCDAppClientCounter struct {
```

Comment:
I think there is something I don't understand here: why is this called `Counter`? Is that because we're counting how many apps we sync?

Reply:
The idea of this mock is to mimic what in Java or Python would be `assertCalledWith()` or similar - I want to make sure that `Sync()` is called with a specific app name. 


I think I called it `Counter` because initially I was just counting number of calls - but that's wrong, we want to know that a specific app was synced, rather than how many times apps where synced. So perhaps there's a better name for it?

Reply:
> The idea of this mock is to mimic what in Java or Python would be assertCalledWith() or similar - I want to make sure that Sync() is called with a specific app name.

Thanks, now it make more sense to me? Maybe `mockArgoCDAppClientWithApp`?